### ISSUE TITLE
pkgsCross.{armv7l-hf,aarch64}-multiplatform.glibc fails to build 

### ISSUE THREAD

**Issue by flokli** [NixOS/nixpkgs#77504]
**To Reproduce**
1. `nix-build -A pkgsCross.aarch64-multiplatform.glibc`
2. `nix-build -A pkgsCross.aarch64-multiplatform.glibc`

**Expected behavior**
The package to successfully build

**Additional context**
```
aarch64-unknown-linux-gnu-gcc ../sysdeps/aarch64/start.S -c     -I../include -I/build/build/csu  -I/build/build  -I../sysdeps/unix/sysv/linux/aarch64  -I../sysdeps/aarch64/nptl  -I../sysdeps/unix/sysv/linux/generic  -I../sysdeps/unix/sysv/linux/wordsize-64  -I../sysdeps/unix/sysv/linux/include -I../sysdeps/unix/sysv/linux  -I../sysdeps/nptl  -I../sysdeps/pthread  -I../sysdeps/gnu  -I../sysdeps/unix/inet  -I../sysdeps/unix/sysv  -I../sysdeps/unix  -I../sysdeps/posix  -I../sysdeps/aarch64/fpu  -I../sysdeps/aarch64/multiarch  -I../sysdeps/aarch64  -I../sysdeps/wordsize-64  -I../sysdeps/ieee754/ldbl-128  -I../sysdeps/ieee754/dbl-64/wordsize-64  -I../sysdeps/ieee754/dbl-64  -I../sysdeps/ieee754/flt-32  -I../sysdeps/aarch64/soft-fp  -I../sysdeps/ieee754  -I../sysdeps/generic  -I.. -I../libio -I. -nostdinc -isystem /nix/store/a2rl0q0ixh6qkdpx9hgy885ijgf27mjz-aarch64-unknown-linux-gnu-stage-static-gcc-debug-9.2.0/lib/gcc/aarch64-unknown-linux-gnu/9.2.0/include -isystem /nix/store/a2rl0q0ixh6qkdpx9hgy885ijgf27mjz-aarch64-unknown-linux-gnu-stage-static-gcc-debug-9.2.0/lib/gcc/aarch64-unknown-linux-gnu/9.2.0/include-fixed -isystem /nix/store/spyz3qksgicas1hnfdfflijq5kg6qlid-linux-headers-4.19.16/include  -D_LIBC_REENTRANT -include /build/build/libc-modules.h -DMODULE_NAME=libc -include ../include/libc-symbols.h  -DPIC     -DTOP_NAMESPACE=glibc -DASSEMBLER  -I/build/build/csu/. -g -Werror=undef -Wa,--noexecstack   -o /build/build/csu/start.o -MD -MP -MF /build/build/csu/start.o.dt -MT /build/build/csu/start.o
In file included from <command-line>:
./../include/libc-symbols.h:471:26: error: '__EI___errno_location' specifies less restrictive attributes than its target '__errno_location': 'const', 'nothrow' [-Werror=missing-attributes]
  471 |   extern __typeof (name) __EI_##name \
      |                          ^~~~~
./../include/libc-symbols.h:475:29: note: in expansion of macro '__hidden_ver1'
  475 | #  define hidden_def(name)  __hidden_ver1(__GI_##name, name, name);
      |                             ^~~~~~~~~~~~~
./../include/libc-symbols.h:539:32: note: in expansion of macro 'hidden_def'
  539 | # define libc_hidden_def(name) hidden_def (name)
      |                                ^~~~~~~~~~
errno-loc.c:28:1: note: in expansion of macro 'libc_hidden_def'
   28 | libc_hidden_def (__errno_location)
      | ^~~~~~~~~~~~~~~
errno-loc.c:24:1: note: '__EI___errno_location' target declared here
   24 | __errno_location (void)
      | ^~~~~~~~~~~~~~~~
cc1: all warnings being treated as errors
make[2]: *** [../o-iterator.mk:9: /build/build/csu/errno-loc.os] Error 1
make[2]: *** Waiting for unfinished jobs....
make[2]: Leaving directory '/build/glibc-2.27/csu'
make[1]: *** [Makefile:215: csu/subdir_lib] Error 2
make[1]: Leaving directory '/build/glibc-2.27'
make: *** [Makefile:9: all] Error 2
builder for '/nix/store/lqjkj6dc17xbvaclamd12b454q6iimnl-glibc-2.27-aarch64-unknown-linux-gnu.drv' failed with exit code 2
cannot build derivation '/nix/store/840ds4ny81z9sfycr5yypmkq597fh0ps-glibc-2.27-aarch64-unknown-linux-gnu.drv': 1 dependencies couldn't be built
error: build of '/nix/store/840ds4ny81z9sfycr5yypmkq597fh0ps-glibc-2.27-aarch64-unknown-linux-gnu.drv' failed
```

**Metadata**
```
 - system: `"x86_64-linux"`
 - host os: `Linux 5.4.6, NixOS, 20.03.git.76e5ae7ebae (Markhor)`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.3.2`
 - channels(root): `"nixpkgs-19.03.172866.4649b6ef4b5"`
 - channels(flokli): `"nixpkgs-20.03pre204199.3140fa89c51"`
 - nixpkgs: `/home/flokli/.nix-defexpr/channels/nixpkgs`
```

**Comment by flokli** [comment]
Seems to be caused by `8f729c0070ec3f78edadeaebcbd110257fe4577e`, at least it seems to have worked with the commit before that.

cc @globin

**Comment by flokli** [comment]
PR: https://github.com/NixOS/nixpkgs/pull/68029

**Comment by Ma27** [comment]
So it seems as this is a new compiler-flag introduced in GCC9. Currently checking if it's possible to build those packages locally with another `-Wno-error` (if that's true, I'd file a PR against staging).

**Comment by thefloweringash** [comment]
See also https://github.com/NixOS/nixpkgs/pull/76972 and https://github.com/NixOS/nixpkgs/pull/76880

**Comment by Ma27** [comment]
Thanks a lot for the note! I basically tried the same what you did and it seems to work, so at least it's tested twice now :sweat_smile:

**Comment by flokli** [comment]
fixed by #76972.
